drivers/storage/portworx/portworx.go

```diff
@@ -168,6 +182,239 @@ func (p *portworx) DeleteStorage(storageCluster *corev1alpha1.StorageCluster) (*
 	}, nil
 }
 
+func (p *portworx) UpdateStorageClusterStatus(
+	cluster *corev1alpha1.StorageCluster,
+) error {
+	clientConn, err := p.getPortworxClient(cluster)
+	if err != nil {
+		return err
+	}
+
+	clusterClient := api.NewOpenStorageClusterClient(clientConn)
+	pxCluster, err := clusterClient.InspectCurrent(context.TODO(), &api.SdkClusterInspectCurrentRequest{})
+	if err != nil {
+		p.sdkConn = nil
+		return fmt.Errorf("failed to inspect cluster: %v", err)
+	} else if pxCluster.Cluster == nil {
+		return fmt.Errorf("empty ClusterInspect response")
+	}
+
+	cluster.Status.Phase = string(mapClusterStatus(pxCluster.Cluster.Status))
+	cluster.Status.ClusterName = pxCluster.Cluster.Name
+	cluster.Status.ClusterUID = pxCluster.Cluster.Id
+
+	return p.updateNodeStatuses(clientConn, cluster)
+}
+
+func (p *portworx) updateNodeStatuses(
+	clientConn *grpc.ClientConn,
+	cluster *corev1alpha1.StorageCluster,
+) error {
+	nodeClient := api.NewOpenStorageNodeClient(clientConn)
+	nodeEnumerateResponse, err := nodeClient.Enumerate(context.TODO(), &api.SdkNodeEnumerateRequest{})
+	if err != nil {
+		return fmt.Errorf("failed to enumerate nodes: %v", err)
+	}
+
+	ownerRef := metav1.NewControllerRef(cluster, controllerKind)
+	for _, nodeID := range nodeEnumerateResponse.NodeIds {
+		nodeResp, err := nodeClient.Inspect(context.TODO(), &api.SdkNodeInspectRequest{NodeId: nodeID})
+		if err != nil {
+			logrus.Warnf("Failed to inspect node %v: %v", nodeID, err)
+			continue
+		} else if nodeResp.Node == nil {
+			logrus.Warnf("Empty NodeInspect response for nodeID %v: %v", nodeID, err)
```

Comment:
No need to print `err` here, it'll always be `nil`

Reply:
Also, if we get an empty response for a ndoe do we want to set a `StatusUnknown` status instead of not listing it at all?

Reply:
Now using Enumerate which gets all nodes. Won't be needing to update status to Unknown.